### PR TITLE
test(dialog): add tests to dialog suite

### PR DESCRIPTION
**Pull request by DlgSHi**
### Checklist

<del> - [ ] Release notes (Conventional Commits)  </del>
<del> - [ ] Unit tests </del>
- [x] Cypress automation tests

<del> - [ ] Storybook added or updated </del>
<del> - [ ] Theme support </del>
<del> - [ ] Typescript `d.ts` file added or updated </del>

### Testing instructions

run `./node_modules/.bin/cypress run --spec './cypress/features/regression/dialog*' --reporter spec --browser chrome`

### CODE REVIEW

File: cypress/features/regression/dialog.feature [hunk 1]
```diff
@@ -72,66 +68,57 @@ Feature: Dialog component
 
   @positive
   Scenario: ShowCloseIcon can close Dialog
```
Comment (KatarzynaQA) [inline cypress/features/regression/dialog.feature]: missing `When`

File: cypress/features/regression/dialog.feature [hunk 2]
```diff
@@ -72,66 +68,57 @@ Feature: Dialog component
...
-      And I hit ESC key
+    Given I check disableEscKey checkbox
+    When I hit ESC key
     Then Dialog is visible
 
   @positive
   Scenario: Enable escape key
-    When I check disableEscKey checkbox
+    Given I check disableEscKey checkbox
       And I uncheck disableEscKey checkbox
```
Comment (KatarzynaQA) [inline cypress/features/regression/dialog.feature]: Change `And` to `When`
```suggestion
      When I uncheck disableEscKey checkbox
```

File: cypress/features/regression/dialog.feature [hunk 3]
```diff
@@ -72,66 +68,57 @@ Feature: Dialog component
...
-      And I open component preview
-      And I hit ESC key
+    When I hit ESC key
```
Comment (KatarzynaQA) [inline cypress/features/regression/dialog.feature]:
```suggestion
    And I hit ESC key
```

File: cypress/support/step-definitions/dialog-steps.js [hunk 4]
```diff
@@ -37,9 +36,23 @@ Then('stickyFormFooter is disabled', () => {
...
-  cy.iFrame(STORY_ROOT).click(position, { force: true });
+  storyRoot().click(position, { force: true });
 });
 
 When('I click on background {string} outside dialog', (position) => {
   backgroundUILocator().click(position, { force: true });
 });
+
+Then('footer buttons have color {string} and has {int} px border', (color, px) => {
+  dialogStickyFormFooter().children().eq(0).children()
```
Comment (KatarzynaQA) [inline cypress/support/step-definitions/dialog-steps.js]: Maybe better will add new variable e.g 'footerButtons' with `dialogStickyFormFooter().children().eq(0).children()` and use it here. It'll be more readable

File: cypress/support/step-definitions/dialog-steps.js [hunk 5]
```diff
@@ -37,9 +36,23 @@ Then('stickyFormFooter is disabled', () => {
...
+    .should('have.css', 'border', `${px}px solid ${color}`)
+    .and('have.css', 'color', color);
+  dialogStickyFormFooter().children().eq(1).children()
```
Comment (KatarzynaQA) [inline cypress/support/step-definitions/dialog-steps.js]: as above

File: cypress/support/step-definitions/dialog-steps.js [hunk 6]
```diff
@@ -37,9 +36,23 @@ Then('stickyFormFooter is disabled', () => {
...
+    .children()
+    .should('have.css', 'background')
+    .should('contain', color);
+  dialogStickyFormFooter().children().eq(1).children()
```
Comment (KatarzynaQA) [inline cypress/support/step-definitions/dialog-steps.js]: as above

File: cypress/features/regression/dialog.feature (hunk 1 above)
Comment (DlgSHi) [reply cypress/features/regression/dialog.feature]: fixed

File: cypress/features/regression/dialog.feature (hunk 2 above)
Comment (DlgSHi) [reply cypress/features/regression/dialog.feature]: fixed

File: cypress/features/regression/dialog.feature (hunk 3 above)
Comment (DlgSHi) [reply cypress/features/regression/dialog.feature]: fixed

File: cypress/support/step-definitions/dialog-steps.js (hunk 4 above)
Comment (DlgSHi) [reply cypress/support/step-definitions/dialog-steps.js]: updated

File: cypress/support/step-definitions/dialog-steps.js (hunk 5 above)
Comment (DlgSHi) [reply cypress/support/step-definitions/dialog-steps.js]: updated

File: cypress/support/step-definitions/dialog-steps.js (hunk 6 above)
Comment (DlgSHi) [reply cypress/support/step-definitions/dialog-steps.js]: updated

File: cypress/support/step-definitions/dialog-steps.js [hunk 7]
```diff
@@ -37,9 +36,21 @@ Then('stickyFormFooter is disabled', () => {
...
   backgroundUILocator().click(position, { force: true });
 });
+
+Then('footer buttons have color {string} and has {int} px border', (color, px) => {
+  dialogStickyFormFooterButton(0)
+    .should('have.css', 'border', `${px}px solid ${color}`)
+    .and('have.css', 'color', color);
+  dialogStickyFormFooterButton(1).children()
+    .should('have.css', 'background')
+    .should('contain', color);
```
Comment (KatarzynaQA) [inline cypress/support/step-definitions/dialog-steps.js]: replace the second `should` to `.and`

File: cypress/support/step-definitions/dialog-steps.js [hunk 8]
```diff
@@ -37,9 +36,21 @@ Then('stickyFormFooter is disabled', () => {
...
+  storyRoot().click(position, { force: true });
 });
 
 When('I click on background {string} outside dialog', (position) => {
   backgroundUILocator().click(position, { force: true });
 });
+
+Then('footer buttons have color {string} and has {int} px border', (color, px) => {
+  dialogStickyFormFooterButton(0)
+    .should('have.css', 'border', `${px}px solid ${color}`)
```
Comment (KatarzynaQA) [inline cypress/support/step-definitions/dialog-steps.js]: move back this line one more upper
dialogStickyFormFooterButton(0).should('have.css', 'border', `${px}px solid ${color}`)

File: cypress/support/step-definitions/dialog-steps.js [hunk 9]
```diff
@@ -37,9 +36,21 @@ Then('stickyFormFooter is disabled', () => {
...
+    .and('have.css', 'color', color);
+  dialogStickyFormFooterButton(1).children()
+    .should('have.css', 'background')
```
Comment (KatarzynaQA) [inline cypress/support/step-definitions/dialog-steps.js]: as above

File: cypress/support/step-definitions/dialog-steps.js [hunk 10]
```diff
@@ -37,9 +36,21 @@ Then('stickyFormFooter is disabled', () => {
...
+    .should('contain', color);
+  dialogStickyFormFooterButton(1).children()
+    .should('have.css', 'border-color', 'rgba(0, 0, 0, 0)')
```
Comment (KatarzynaQA) [inline cypress/support/step-definitions/dialog-steps.js]: as above

File: cypress/support/step-definitions/dialog-steps.js (hunk 7 above)
Comment (DlgSHi) [reply cypress/support/step-definitions/dialog-steps.js]: fixed

File: cypress/support/step-definitions/dialog-steps.js (hunk 8 above)
Comment (DlgSHi) [reply cypress/support/step-definitions/dialog-steps.js]: fixed

File: cypress/support/step-definitions/dialog-steps.js (hunk 9 above)
Comment (DlgSHi) [reply cypress/support/step-definitions/dialog-steps.js]: fixed

File: cypress/support/step-definitions/dialog-steps.js (hunk 10 above)
Comment (DlgSHi) [reply cypress/support/step-definitions/dialog-steps.js]: fixed

**Pull request merged**